File: android/app/src/main/kotlin/com/stockbook/app/design/Icons.kt
package com.stockbook.app.design

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Inventory2
import androidx.compose.material.icons.filled.KeyboardArrowLeft
import androidx.compose.material.icons.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.LibraryAdd
import androidx.compose.material.icons.filled.LightMode
import androidx.compose.material.icons.filled.Payments
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Receipt
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Share
import androidx.compose.material.icons.filled.UnfoldMore
import androidx.compose.material.icons.outlined.AddCircle
import androidx.compose.material.icons.outlined.Home
import androidx.compose.material.icons.outlined.Inventory2
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.Receipt
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * Every glyph the app uses, in one place.
 *
 * The design calls for **Phosphor Icons**, which are not on the system, so each
 * Phosphor name maps to its nearest Material equivalent *here* — exactly as the
 * iOS build maps them to SF Symbols. Adding the real Phosphor set changes this
 * file and nothing else.
 */
object Icon {
    // Navigation / chrome
    val settings: ImageVector = Icons.Filled.Settings          // phosphor: gear-six
    val today: ImageVector = Icons.Outlined.Home               // phosphor: house
    val todayActive: ImageVector = Icons.Filled.Home
    val items: ImageVector = Icons.Outlined.Inventory2         // phosphor: shapes
    val itemsActive: ImageVector = Icons.Filled.Inventory2
    val sell: ImageVector = Icons.Outlined.AddCircle           // phosphor: plus-circle
    val sellActive: ImageVector = Icons.Filled.Add
    val bills: ImageVector = Icons.Outlined.Receipt            // phosphor: receipt
    val billsActive: ImageVector = Icons.Filled.Receipt
    val people: ImageVector = Icons.Outlined.Person            // phosphor: user
    val peopleActive: ImageVector = Icons.Filled.Person
    val back: ImageVector = Icons.AutoMirrored.Filled.ArrowBack

    // Actions
    val add: ImageVector = Icons.Filled.Add
    val remove: ImageVector = Icons.Filled.Remove
    val delete: ImageVector = Icons.Filled.Delete
    val edit: ImageVector = Icons.Filled.Edit
    val addStock: ImageVector = Icons.Filled.LibraryAdd      // phosphor: stack-plus
    val close: ImageVector = Icons.Filled.Close
    val confirm: ImageVector = Icons.Filled.Check
    val openRow: ImageVector = Icons.Filled.KeyboardArrowRight
    // Stepping one day at a time. Distinct from `back`, which leaves a screen —
    // these stay on it and change what it is showing.
    val stepBack: ImageVector = Icons.Filled.KeyboardArrowLeft  // phosphor: caret-left
    val stepForward: ImageVector = Icons.Filled.KeyboardArrowRight // phosphor: caret-right
    val chooseFromList: ImageVector = Icons.Filled.UnfoldMore
    val share: ImageVector = Icons.Filled.Share            // phosphor: share-network

    // People & money
    val customer: ImageVector = Icons.Filled.Person
    /**
     * Money changing hands: the owed banner, and the quick action that opens it.
     *
     * Banknotes rather than a currency mark. Every `$` glyph in an icon set is
     * wrong for a shop billing in riyals, and the two Material ones that carry a
     * sign say the wrong thing in the one place this is drawn.
     */
    val owed: ImageVector = Icons.Filled.Payments             // phosphor: hand-coins

    /** The owner's own spending. */
    val expenses: ImageVector = Icons.Outlined.Receipt

    // Appearance
    val themeDark: ImageVector = Icons.Filled.DarkMode        // phosphor: moon
    val themeLight: ImageVector = Icons.Filled.LightMode      // phosphor: sun
}

File: ios/Stockbook/DesignSystem/Icons.swift
import SwiftUI

/// Every glyph the app uses, in one place.
///
/// The design calls for **Phosphor Icons** (regular weight, filled only for the
/// active tab and the backup shield). Phosphor is not on the system, so each
/// name below maps to the closest SF Symbol. When the Phosphor font or an asset
/// set is added to the project, this is the only file that changes — no feature
/// code names a glyph directly.
enum Icon {
    // Navigation / chrome
    static let settings = "gearshape"                 // phosphor: gear-six
    static let today = "house"                        // phosphor: house
    static let todayActive = "house.fill"
    static let items = "shippingbox"                  // phosphor: shapes
    static let itemsActive = "shippingbox.fill"
    static let sell = "plus.circle"                   // phosphor: plus-circle
    static let sellActive = "plus.circle.fill"
    static let bills = "doc.text"                     // phosphor: receipt
    static let billsActive = "doc.text.fill"
    static let back = "arrow.left"                    // phosphor: arrow-left

    // Actions
    static let add = "plus"                           // phosphor: plus
    static let remove = "minus"                       // phosphor: minus
    static let delete = "trash"                       // phosphor: trash
    static let edit = "pencil"                        // phosphor: pencil-simple
    static let close = "xmark"                        // phosphor: x
    static let confirm = "checkmark"                  // phosphor: check
    static let browseAll = "text.magnifyingglass"     // phosphor: list-magnifying-glass
    static let addStock = "plus.square.on.square"     // phosphor: stack-plus
    static let chooseFromList = "chevron.up.chevron.down" // phosphor: caret-up-down
    static let openRow = "chevron.right"              // phosphor: caret-right

    // People & money
    static let customer = "person"                    // phosphor: user
    static let owed = "banknote"                      // phosphor: hand-coins
    static let money = "dollarsign.circle"            // phosphor: money
    /// The owner's own spending. A receipt, not a coin: what is written down
    /// here is the slip, and the coin icons are already spoken for.
    static let expenses = "receipt"                   // phosphor: receipt
    static let partPayment = "scissors"               // phosphor: scissors

    // Appearance
    static let themeDark = "moon"                     // phosphor: moon
    static let themeLight = "sun.max"                 // phosphor: sun

    // Files & backup
    static let export = "square.and.arrow.up"         // phosphor: export
    static let importFile = "tray.and.arrow.down"     // phosphor: tray-arrow-down
    static let file = "doc.text"                      // phosphor: file-text
    static let share = "square.and.arrow.up"          // phosphor: share-network
    static let folder = "folder"                      // phosphor: folder-open
    static let backupMissing = "exclamationmark.shield" // phosphor: shield-warning
    static let backupDone = "checkmark.shield.fill"   // phosphor: shield-check (filled)
}

/// A glyph at an explicit point size, the way the design specifies them.
struct Glyph: View {
    let name: String
    var size: CGFloat = 16

    init(_ name: String, size: CGFloat = 16) {
        self.name = name
        self.size = size
    }

    var body: some View {
        Image(systemName: name)
            .font(.system(size: size, weight: .regular))
    }
}
